src/lib/gssapi/krb5/init_sec_context.c

```diff
@@ -582,8 +587,18 @@ kg_new_connection(
                                   &ctx->there)))
         goto cleanup;
 
-    code = get_credentials(context, cred, ctx->there, now,
-                           ctx->krb_times.endtime, &k_cred);
+    if (cred->impersonator) {
```

Comment:
Actually, on second thought, mutating a local `flags` here is silly, so just:

    if (cred->impersonator)
        code = get_credentials(context, cred, ctx->there, now, ctx->krb_times.endtime, &k_cred, KRB5_GC_CACHED);
    if (code == KRB5_CC_NOTFOUND)
        code = get_credentials(context, cred, ctx->there, now, ctx->krb_times.endtime, &k_cred, 0);
    if (code)
        goto cleanup;